pypit/calibrations.py

```diff
@@ -0,0 +1,627 @@
+                                  det=self.det, setup=self.setup,
+                                  tslits_dict=self.tslits_dict, settings_det=self.settings['detector'],
+                                  pixlocn=self.pixlocn)
+            # Master
+            self.mstilts = self.waveTilts.master()
+            if self.mstilts is None:
+                self.mstilts, self.wt_maskslits = self.waveTilts.run(maskslits=self.maskslits,
+                                                      wv_calib=self.wv_calib, doqa=self.write_qa)
+                if self.save_masters:
+                    self.waveTilts.save_master()
+            else:
+                self.wt_maskslits = np.zeros_like(self.maskslits, dtype=bool)
+            # Save
+            self.calib_dict[self.setup]['tilts'] = self.mstilts
+            self.calib_dict[self.setup]['wtmask'] = self.wt_maskslits
+
+        # Mask me
+        self.maskslits += self.wt_maskslits
+
+        # Return
+        return self.mstilts, self.maskslits
+
+    def run_the_steps(self):
+        for step in self.steps:
+            getattr(self, 'get_{:s}'.format(step))()
+
+
+    def _chk_set(self, items):
+        for item in items:
+            if getattr(self, item) is None:
+                msgs.error("Use self.set to specify '{:s}' prior to generating XX".format(item))
+
+    def _chk_objs(self, items):
+        for obj in items:
+            if getattr(self, obj) is None:
+                msgs.warn("You need to generate {:s} prior to this calibration..".format(obj))
+                if obj in ['tslits_dict','maskslits']:
+                    msgs.warn("Use get_slits".format(obj))
+                else:
+                    # Strip ms
+                    if obj[0:2] == 'ms':
+                        iobj = obj[2:]
+                    else:
+                        iobj = obj
+                    msgs.warn("Use get_{:s}".format(iobj))
+                return False
+        return True
+
+    def show(self, obj):
+        if isinstance(obj, np.ndarray):
+            if len(obj.shape) == 2:
+                debugger.show_image(obj)
+        else:
+            msgs.warn("Not ready for this type of object")
+
+    def __repr__(self):
+        # Generate sets string
+        txt = '<{:s}: setup={}, det={}, sci_ID={}'.format(self.__class__.__name__,
+                                                          self.setup,
+                                                          self.det,
+                                                          self.sci_ID)
+        txt += '>'
+        return txt
+
+class MultiSlitCalibrations(Calibrations):
+
+    def __init__(self, fitstbl, steps=None, **kwargs):
+        # Get started
+        Calibrations.__init__(self, fitstbl, **kwargs)
+
+        # Standard steps
+        if steps is None:
+            self.steps = ['datasec_img', 'bias', 'arc', 'bpm', 'pixlocn', 'slits', 'wv_calib', 'tilts', 'pixflatnrm', 'wave']
```

Comment:
By the way, this felt like a good movie... it's only at the end of the file when everything comes together.  I hadn't expected this, but I think this is a great touch, and will allow us to develop the code to reduce all kinds of data. Suitably impressed...   :-)

Reply:
Thanks!